### ISSUE TITLE
Initialize Hydra using files from a previous run

### ISSUE THREAD

**Issue by edornd** [facebookresearch/hydra#1576]
Hi,

This is more like a question than an issue, but I didn't really know where to ask, sorry about that.

I am currently using Hydra with Pytorch Lightning for training and it's going great so far: I can dynamically instantiate the modules I need, while Hydra keeps the output dir clean and organized.

My output directory tree looks more or less like this:

```
outputs/
    ${now:%Y-%m-%d_%H-%M-%S}/
        .hydra/ ... hydra config files
        checkpoints/
        run.log
        (other non-relevant stuff)
```
I have the following two use cases, for which I found a possible workaround, but just to avoid XY problems I'm going for the direct questions:
1) I'd like to run a prediction/test run on an old run: given the `.hydra` dir, I was hoping there was a way to basically say "take the same configuration" + some optional override, since datasets, models and other configuration is exactly what I'd need to declare anyway and I certainly do not want another output folder for a simple test.
2) Quite similar to the previous question: I'd also like to resume a training from a previous run, again using 90-95% of the previous configuration, minus some overrides (e.g. step=2, lr=...)

At the moment, I managed to assemble an ugly yet working workaround, exploiting the suggestions from #1022:
``` python
    initialize(config_path=os.path.join(path, ".hydra"))
    old_cfg = compose(config_name="config.yaml", return_hydra_config=False)
    print(OmegaConf.to_yaml(old_cfg, resolve=True))
```

I honestly don't like this setup a lot since I'm forced to keep 'hydra_config' to false, otherwise, given the missing HydraConfig, I cannot resolve the interpolation variables in the OmegaConf. Other than that, I'd say I'm fine with it.

The TLDR is: is this the correct approach, or is there a better way for what I have in mind?
Thanks!

**Comment by omry** [comment]
I recently answered a [similar question](https://stackoverflow.com/questions/67170653/how-to-load-hydra-parameters-from-previous-jobs-without-having-to-use-argparse/67172466?noredirect=1) on StackOverflow.

I think the suggestion solution there will cover you to a large extent.

**Comment by omry** [comment]
About HydraConfig, you will need to initialize it manually.

In general it doesn't make sense when used with compose because Hydra is not performing any actions exception the composition.
You can still initialize the HydraConfig singleton if you really want to, but what you need from it may not be there.


```python
# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
from dataclasses import dataclass
from typing import Any

from hydra.core.config_store import ConfigStore
from hydra.core.hydra_config import HydraConfig
from omegaconf import OmegaConf, open_dict

from hydra import compose, initialize


@dataclass
class Config:
    overrides: Any = "${hydra:overrides.task}"


ConfigStore.instance().store("config", node=Config)

if __name__ == "__main__":
    initialize(config_path=None)
    # return_hydra_config=True
    cfg = compose("config", return_hydra_config=True, overrides=["+a=10", "+b=20"])
    # initialize the HydraConfig singleton:
    HydraConfig.instance().set_config(cfg)
    # Optional: remove the hydra config node from the used config.
    with open_dict(cfg):
        del cfg["hydra"]
    print(OmegaConf.to_yaml(cfg, resolve=True))
```

Output:
```
overrides:
- +a=10
- +b=20
a: 10
b: 20
```

**Comment by edornd** [comment]
Thank you very much, this is indeed a weird use case, I am aware of that. Not having an in-depth knowledge about the library also doesn't help, but I I should be able to assemble a cleaner and functioning test loop with this info! 

For the HydraConfig, I guess I kind of solved it: the problem was linked to some weird interpolations I had on some modules (a logger had as extra description `hydra:override_dirname`, the very same old hydra config has `hydra:job.num` that couldn't be interpolated).
I simply removed those nodes, since I didn't really need the logger anyway, same goes for the old hydra params. Cool to know that there's a quick way to also solve this in a cleaner way, I might have a look at that in the future.

Again, thanks a lot for the quick response!

**Comment by omry** [comment]
Great.
hydra.job.num is another thing that does not make sense at all when using compose.
what job, what num?
you can also override those from the when composing, providing some values to them.

**Comment by addisonklinke** [comment]
> I recently answered a [similar question](https://stackoverflow.com/questions/67170653/how-to-load-hydra-parameters-from-previous-jobs-without-having-to-use-argparse/67172466?noredirect=1) on StackOverflow

Thanks for the tip @omry. Do you see this eventually making its way into a core library feature, or will it likely be left up to users to implement that pattern?

**Comment by Jasha10** [comment]
@addisonklinke we are working on some features to support use-cases related to reproducing previous Hydra runs. See the recent activity on #1805.
